services/auth/oauth2.go

```diff
@@ -142,6 +179,15 @@ func (o *OAuth2) Verify(req *http.Request, w http.ResponseWriter, store DataStor
 		return nil, nil
 	}
 
+	user, err := parseActionsJWT(req, store)
```

Comment:
I believe it should improve `parseToken` but not introduce a new function with duplicate/inconsistent logic.

Reply:
Is something like https://github.com/go-gitea/gitea/pull/32527/commits/6e876cc9c630219f892edae5bf7a6a364392b863 what you had in mind?

Reply:
It looks better, but where is the permission check? repo1's action token could be used in repo2?

Reply:
This is just Authentication. Authorization happens with the provided `ActionsTaskID` (e.g. https://github.com/go-gitea/gitea/blob/0d5abd9b3e04a09f5d7de720c99e3451723e028e/routers/api/v1/api.go#L192C24-L192C37). The format of Authentication happens the same here as the previous token format.

Reply:
Thank you for the clarification.

The last question: is it possible to write some test code (unit test should be enough) for the new logic?

Reply:
I added some tests in https://github.com/go-gitea/gitea/pull/32527/commits/f5c0b53d55b9c14b12bd0ba7228f656654e22ac9. Are these what you had in mind?